File: source/java/src/com/formationds/commons/model/Service.java
/*
 * Copyright (c) 2014, Formation Data Systems, Inc. All Rights Reserved.
 */

package com.formationds.commons.model;

import com.formationds.commons.model.abs.ModelBase;
import com.formationds.commons.model.type.ManagerType;
import com.formationds.commons.model.type.ServiceStatus;

/**
 * @author ptinius
 */
public class Service
  extends ModelBase {
  private static final long serialVersionUID = -1577170593630479004L;

  private Long uuid;

  private String autoName;
  private Integer port;
  private ServiceStatus status;
  private ManagerType type;

  protected Service( ) {
  }

  public String getAutoName( ) {
    return autoName;
  }
  
  public void setAutoName( String aName ){
	  this.autoName = aName;
  }

  public Long getUuid( ) {
    return uuid;
  }
  
  public void setUuid( Long aUuid ){
	  this.uuid = aUuid;
  }

  public ManagerType getType( ) {
    return type;
  }
  
  public void setType( ManagerType aType ){
	  this.type = aType;
  }

  public ServiceStatus getStatus( ) {
    return status;
  }
  
  public void setStatus( ServiceStatus aStatus ){
	  this.status = aStatus;
  }

  public Integer getPort( ) {
    return port;
  }
  
  public void setPort( Integer aPort ){
	  this.port = aPort;
  }

}

File: source/cli/src/fds/model/service.py

class Service(object):
    '''
    Created on Apr 28, 2015
    
    @author: nate
    '''

    def __init__(self, status="UNKNOWN", port=0, auto_name="UNKNOWN", a_type="FDSP_PLATFORM", an_id=-1):
        self.__status = status
        self.__port = port
        self.__auto_name = auto_name
        self.__id = an_id
        self.__type = a_type
        return
    
    @property
    def auto_name(self):
        return self.__auto_name
    
    @auto_name.setter
    def auto_name(self, a_name):
        self.__auto_name = a_name
        
    @property
    def port(self):
        return self.__port
    
    @port.setter
    def port(self, port):
        self.__port = port
        
    @property
    def status(self):
        return self.__status
    
    @status.setter
    def status(self, status):
        self.__status = status
      
    @property  
    def type(self):
        return self.__type
    
    @type.setter
    def type(self, a_type):
        self.__type = a_type
        
    @property
    def id(self):
        return self.__id
    
    @id.setter
    def id(self, an_id):
        self.__id = an_id